test/unit/app/renderer/downloadItemTest.js

```diff
@@ -119,16 +122,39 @@ describe('downloadItem component', function () {
       })
 
       testButton('.deleteButton', [downloadStates.CANCELLED, downloadStates.INTERRUPTED, downloadStates.COMPLETED], function (button) {
-        const spy = sinon.spy(appActions, 'downloadDeleted')
-        const buttonVisibilitySpy = sinon.spy(appActions, 'showDownloadDeleteConfirmation')
-        assert.equal(this.result.find('.confirmDeleteButton').length, 0)
-        button.simulate('click')
-        assert(buttonVisibilitySpy.called)
-        this.result.find('.confirmDeleteButton').simulate('click')
-        assert(spy.withArgs(this.downloadId).calledOnce)
-        appActions.downloadDeleted.restore()
-        appActions.showDownloadDeleteConfirmation.restore()
+        const spy = sinon.spy(appActions, 'showDownloadDeleteConfirmation')
+        try {
+          // Confirmation should NOT be visible by default
+          assert.equal(this.result.find('.confirmDeleteButton').length, 0)
+
+          // Clicking delete should show confirmation
+          button.simulate('click')
+          assert(spy.called)
+        } finally {
+          appActions.showDownloadDeleteConfirmation.restore()
+        }
       })
+
+      if ([downloadStates.CANCELLED, downloadStates.INTERRUPTED, downloadStates.COMPLETED].includes(state)) {
+        describe('when delete button has been clicked', function () {
+          beforeEach(function () {
+            this.result.setProps({
+              deleteConfirmationVisible: true
+            })
+          })
+
+          testButton('.confirmDeleteButton', [downloadStates.CANCELLED, downloadStates.INTERRUPTED, downloadStates.COMPLETED], function (button) {
+            const spy = sinon.spy(appActions, 'downloadDeleted')
+            try {
```

Comment:
In case something fails.  For example, with your commit, it failed because 0 items were found.   This raised an exception which then did not properly restore the sinon spy.  When the next test case ran, it would try to put a spy on the same method but fail (because spy has already been done).  The try is required only because I wanted to use a finally to restore the method which had been spied